test/modules/generic-mqtt-tester/src/message_channel.rs

```diff
@@ -49,27 +56,32 @@ impl MessageHandler for ReportResultMessageHandler {
         received_publication: ReceivedPublication,
     ) -> Result<(), MessageTesterError> {
         let sequence_number = parse_sequence_number(&received_publication);
-        let batch_id = Uuid::from_u128_le(received_publication.payload.slice(4..20).get_u128_le());
-
-        info!(
-            "reporting result for publication with sequence number {}",
-            sequence_number,
-        );
-        let result = MessageTestResult::new(
-            self.tracking_id.clone(),
-            batch_id.to_string(),
-            sequence_number,
-        );
-
-        let test_type = trc_client::TestType::Messages;
-        let created_at = chrono::Utc::now();
-
-        if let Err(e) = self
-            .reporting_client
-            .report_result(self.report_source.clone(), result, test_type, created_at)
-            .await
-        {
-            error!("error reporting result to trc: {:?}", e);
+        let received_batch_id =
+            Uuid::from_u128_le(received_publication.payload.slice(4..20).get_u128_le());
+
+        if Some(received_batch_id) == self.batch_id {
+            info!(
+                "reporting result for publication with sequence number {}",
+                sequence_number,
+            );
+            let result = MessageTestResult::new(
+                self.tracking_id.clone(),
+                received_batch_id.to_string(),
+                sequence_number,
+            );
+
+            let test_type = trc_client::TestType::Messages;
+            let created_at = chrono::Utc::now();
+
+            if let Err(e) = self
+                .reporting_client
+                .report_result(self.report_source.clone(), result, test_type, created_at)
+                .await
+            {
+                error!("error reporting result to trc: {:?}", e);
+            }
+        } else {
+            error!("received publication with non-matching batch id")
```

Comment:
Is this really an error? If it happens often and/or is an expected occurrence due, consider lowering the log level.

Reply:
Yeah it should be wrn I think.